reactor-core/src/main/java/reactor/core/publisher/FluxSwitchOnFirst.java

```diff
@@ -0,0 +1,646 @@
+                        "The transformer returned a null value"
+                    );
+                }
+                catch (Throwable e) {
+                    done = true;
+                    Operators.error(o, Operators.onOperatorError(s, e, t, o.currentContext()));
+                    return;
+                }
+
+                first = t;
+                result.subscribe(o);
+                return;
+            }
+
+            i.onNext(t);
+        }
+
+        @Override
+        public void onError(Throwable t) {
+            if (done) {
+                Operators.onErrorDropped(t, currentContext());
+                return;
+            }
+
+            throwable = t;
+            done = true;
+            CoreSubscriber<? super T> i = inner;
+            T f = first;
+
+            if (f == null && i == null) {
+                Publisher<? extends R> result;
+                CoreSubscriber<? super R> o = outer;
+
+                try {
+                    result = Objects.requireNonNull(
+                        transformer.apply(Signal.error(t, o.currentContext()), this),
+                        "The transformer returned a null value"
+                    );
+                }
+                catch (Throwable e) {
+                    done = true;
+                    Operators.error(o, Operators.onOperatorError(s, e, t, o.currentContext()));
+                    return;
+                }
+
+                result.subscribe(o);
+                return;
+            }
+
+            if (f == null) {
+                drainRegular();
+            }
+        }
+
+        @Override
+        public void onComplete() {
+            if (done) {
+                return;
+            }
+
+            done = true;
+            CoreSubscriber<? super T> i = inner;
+            T f = first;
+
+            if (f == null && i == null) {
+                Publisher<? extends R> result;
+                CoreSubscriber<? super R> o = outer;
+
+                try {
+                    result = Objects.requireNonNull(
+                        transformer.apply(Signal.complete(o.currentContext()), this),
+                        "The transformer returned a null value"
+                    );
+                }
+                catch (Throwable e) {
+                    done = true;
+                    Operators.error(o, Operators.onOperatorError(s, e, null, o.currentContext()));
+                    return;
+                }
+
+                result.subscribe(o);
+                return;
+            }
+
+            if (f == null) {
+                drainRegular();
+            }
+        }
+
+        @Override
+        public void request(long n) {
+            if (Operators.validate(n)) {
+                if (once == 1 && ONCE.compareAndSet(this, 1, 2)) {
+                    if (first != null) {
+                        drainRegular();
+                    }
+
+                    if (n != Long.MAX_VALUE && --n > 0) {
```

Comment:
this part seems to never be covered in tests (from IntelliJ coverage stats). Same in the conditional version.

Reply:
nice catchup! found a 🐛 there! 😄